sdk/messaging/azservicebus/internal/errors.go

```diff
@@ -208,6 +198,59 @@ func shouldRecreateConnection(ctxForLogging context.Context, err error) bool {
 	return shouldRecreate
 }
 
+type recoveryKind string
+
+const RecoveryKindNone recoveryKind = ""
+const RecoveryKindFatal recoveryKind = "fatal"
+const RecoveryKindLink recoveryKind = "link"
+const RecoveryKindConn recoveryKind = "connection"
+
+type ServiceBusError struct {
+	inner        error
+	RecoveryKind recoveryKind
+}
+
+func (sbe *ServiceBusError) String() string {
+	return sbe.inner.Error()
+}
+
+func (sbe *ServiceBusError) AsError() error {
```

Comment:
This should be `Unwrap()` so that it seemlessly works with `errors.As()`, see [docs](https://pkg.go.dev/errors).

Reply:
I think I named this badly - it's not an `error` (although it was for a little bit). It's just...my wrapper around error that's not intended to be passed directly. Can you think of a less misleading name?

Reply:
I get it now, I missed that ServiceBusError isn't an actual error.  Maybe ErrorInfo or something like that.

Reply:
(I'll rename this in the other PR to make my merge less gross)